mono/mini/mini-arm.c

```diff
@@ -3085,8 +3085,8 @@ mono_arch_finish_dyn_call (MonoDynCallInfo *info, guint8 *buf)
 	DynCallArgs *p = (DynCallArgs*)buf;
```

Comment:
Could use `HostDynCallArgs`  and `TargetDynCallArgs`.

Reply:
A lot of this code doesn't run during AOTing, so it doesn't need host/target.

Reply:
In terms of my change, right, I think the errors all describe things that will not happen.

But I suggest either my changes, or ifdef cross_compile out a bunch of code.
The compiler doesn't know the code is dead and I'm not sure what else to do about it. Maybe we won't use C++ and C will ignore them. But presently I need SDKs to work with C++ to validate C++.

In terms of DynCallArgs, about same thing, agreed. It seems a bit off though.
If we had HostDynCallArgs and TargetDynCallArgs, we could cast the constituents less, just to fix these errors, in code that doesn't run.

If we ever have a multiple-configuration runtime, that both crosses word size and runs natively, then something larger is needed.